tests/foreman/api/test_contentmanagement.py

```diff
@@ -88,8 +99,8 @@ def test_positive_sync_repos_with_large_errata(self):
         for _ in range(2):
             product = entities.Product(organization=org).create()
             repo = entities.Repository(product=product, url=FAKE_7_YUM_REPO).create()
-            with self.assertNotRaises(TaskFailedError):
-                repo.sync()
+            response = repo.sync()
```

Comment:
Consider using try/except here to call `pytest.fail` should the call to repo.sync raise TaskFailedError. That way the test result is failure, not error.

Reply:
There is actually a recommendation from pytest docs for consistently dealing with this type of behavior by creating a dummy context manger. 

https://docs.pytest.org/en/stable/example/parametrize.html#parametrizing-conditional-raising

Not for the scope of this PR, but something we could consider adding for all test modules to use.